src/schema/v2/fair.ts

```diff
@@ -262,35 +266,48 @@ export const FairType = new GraphQLObjectType<any, ResolverContext>({
             type: GraphQLBoolean,
             defaultValue: false,
           },
+          page: {
+            type: GraphQLInt,
+            defaultValue: 1,
+          },
         }),
         resolve: ({ id }, options, { fairBoothsLoader }) => {
+          const pageOptions = convertConnectionArgsToGravityArgs(options)
+          const { page, size, offset } = pageOptions
+
           interface GravityOptions {
             size: number
             sort: string
             cursor?: string
             section: string
             artworks: boolean
             total_count: boolean
+            page: number
           }
           const gravityOptions: GravityOptions = {
             sort: options.sort || "-featured",
             section: options.section,
-            size: options.first,
+            size,
             artworks: true,
             total_count: !!options.totalCount,
+            page,
           }
-          if (!!options.after) {
-            gravityOptions.cursor = options.after
-          }
+
           return fairBoothsLoader(id, gravityOptions).then(
-            ({ body: { results, next } }) => {
-              const connection = connectionFromArraySlice(results, options, {
-                arrayLength: results.length,
-                sliceStart: 0,
-              })
-              connection.pageInfo.endCursor = next
-              connection.pageInfo.hasNextPage = !!next
-              return connection
+            ({ body, headers }) => {
+              const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+
+              return {
+                totalCount,
+                pageCursors: createPageCursors(
+                  { ...options, page, size },
```

Comment:
**minor:** I don't believe we need to pass `...options` in here, because we just [take this object and destructure it to access page and size](https://github.com/artsy/metaphysics/blob/3fcdbf0d4c9355802666d6184e6204573a8df01f/src/schema/v2/fields/pagination.ts#L97)

Reply:
Good catch!